README.md

```diff
@@ -1 +1,66 @@
 # snare
+
+`snare` is a simple GitHub webhooks runner. When a request comes in, it runs an
+arbitrary program for that repository informing it of the event -- that program
+can then perform whatever actions it wants.
+
+
+## Basic usage
+
+`snare` requires three command-line arguments to be specified:
+
+```
+Usage: snare -p <port> -r <repo scripts dir> -s <secret>
+```
+
+where:
+
+ * `<port>` is a port number (e.g. 4567)
+ * `<repo scripts dir>` is the directory where programs for each repository
+   targeted by the webhook are stored. For a repository `repo` owned by `user`
+   the command `<repo scripts dir>/<user>_<repo> <event> <path to GitHub JSON>`
+   will be run.
+ * `<secret>` is the GitHub secret which guarantees that hooks are coming from
+   your GitHub repository and not a malfeasant.
+
+The most important part of this are the per-repo programs. For example,
+`snare`'s GitHub repository is
+[`https://github.com/softdevteam/snare`](https://github.com/softdevteam/snare):
+if we set up a web hook up that notifies us of when pull request events have
+been made then the command `<repo scripts dir>/softdevteam_snare pull_request
+/path/to/json` will be executed where `pull_request` is the name of the GitHub
+event and `/path/to/JSON` is a path to a file containing the complete GitHub
+JSON for that event. The `softdevteam_snare` program can then execute whatever
+it wants. In order to work out precisely what event has happened, you will
+need to read (GitHub's webhooks
+documentation)[https://developer.github.com/webhooks/].
+
+
+## Example repository program
+
+If we want to only execute commands when a pull request is merged, our program
+for that repository might look as follows:
+
+```sh
+#! /bin/sh
+
+if [ $1 != "pull_request" ]; then
+    exit 0
+fi
+
+if [ "X`jq .action $2 | tr -d '\"'`" != "Xclosed" ]; then
+    exit 0
+fi
+
+if [ "X`jq .pull_request.merged $2 | tr -d '\"'`" != "Xtrue" ]; then
+    exit 0
+fi
+```
+
+where `jq` is a command-line JSON processor. If all three of those `if`
```

Comment:
Should we mention that `jq` is not just any JSON processor here?

Reply:
People can use whatever they want, but `jq` is one way of doing it. I've linked to the `jq` website so that interested parties can investigate it.